### PR TITLE
Twix Remote Control

### PR DESCRIPTION
**Pull request by knoellle**
## Introduced Changes

Allows twix to act as a remote control for the robot. Currently only supports walking, no kicks.
Tested with my steamdeck but should™ work with any controller.

Based on: #22 

## ToDo / Known Issues

- [x] [Install libudev in CI runners](https://github.com/HULKs/hulk/actions/runs/3615021255/jobs/6091848765)

## Ideas for Next Iterations (Not This PR)

- Add kick support
- Visualization of next step in twix
- Only update parameter when axis value changes
- Add support for controlling with keyboard (WASD, arrowkeys, etc.)

## How to Test

- Connect gamepad to computer
- Start twix and open "Remote" panel
- Connect to nao
- Press the start button on the controller or check the "enabled" checkbox
- Move left/right stick to make the robot walk/turn in that direction.

### CODE REVIEW

File: tools/twix/src/panels/remote.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    gilrs: Gilrs,
+    active_gamepad: Option<GamepadId>,
+    enabled: bool,
+}
+
+impl Panel for RemotePanel {
+    const NAME: &'static str = "Remote";
+
+    fn new(nao: Arc<Nao>, _value: Option<&Value>) -> Self {
+        let gilrs = Gilrs::new().unwrap();
```
Comment (okiwi6) [inline tools/twix/src/panels/remote.rs:21]: maybe you can present a better error message if the `Gilrs` creation failed?

File: tools/twix/src/panels/remote.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        self.nao
+            .update_parameter_value("control.step_planner.injected_step", step);
+    }
+}
+
+impl Widget for &mut RemotePanel {
+    fn ui(self, ui: &mut eframe::egui::Ui) -> eframe::egui::Response {
+        self.gilrs.inc();
+
+        if ui.checkbox(&mut self.enabled, "Enabled (Start)").changed() {
```
Comment (okiwi6) [inline tools/twix/src/panels/remote.rs:53]: Maybe that`s just personal, but for readability I'd prefer
```rs
ui.checkbox(&mut self.enabled, "Enabled (Start)");
if self.enabled { ... }
```

**Review by okiwi6 — CHANGES_REQUESTED** [review]
I think this feature is quite neat. If you have some time, maybe you can rebase it to main, so that I can test it

File: tools/twix/src/panels/remote.rs (hunk 2 above)
Comment (knoellle) [reply tools/twix/src/panels/remote.rs:53]: That's not the same logic though.
The if block is supposed executed when the enabled state changed, not every render cycle when enabled is true.

File: tools/twix/src/panels/remote.rs (hunk 1 above)
Comment (knoellle) [reply tools/twix/src/panels/remote.rs]: Changed to expect with a message explaining what happened.

**Comment by knoellle** [comment]
Rebased and tested. Had to change a parameter path because we no longer prefix them with the cycler name.

**Comment by okiwi6** [comment]
Doesn't work for me:
```
INFO] Try connection to ws://10.1.24.37:1337
[INFO] Connected to ws://10.1.24.37:1337
[INFO] Gamepad /dev/input/event15 (GuliKit Controller XW) connected.
[INFO] Gamepad /dev/input/event15 (GuliKit Controller XW) connected.
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to open /home/ole/Documents/hulk/etc/configuration/hardware_ids.json

Caused by:
    No such file or directory (os error 2)

Location:
    /home/ole/Documents/hulk/crates/repository/src/lib.rs:280:14', tools/twix/src/repository_parameters.rs:26:67
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Review by okiwi6 — DISMISSED** [review]
Now works for me, nice feature

**Review by okiwi6 — APPROVED** [review]
:+1:

**Pull request merged**